app/models/user_access.rb

```diff
@@ -70,18 +70,40 @@ def accessible_facilities(action)
       .includes(facility_group: :organization)
   end
 
+  def accessible_district_regions(action)
+    facility_group_ids = accessible_facility_groups(action).pluck("facility_groups.id")
+    Region.district_regions.where(source_id: facility_group_ids)
+  end
+
+  def accessible_block_regions(action)
+    facility_group_ids = accessible_facility_groups(action).pluck("facility_groups.id")
+    paths = Region.district_regions.where(source_id: facility_group_ids, source_type: "FacilityGroup").pluck(:path)
+    globs = paths.map { |path| " '#{path}.*' "}.join(",")
+    Region.block_regions.where("path ? ARRAY[#{globs}]::lquery[]")
+  end
+
+  def accessible_facility_regions(action)
+    facility_ids = accessible_facilities(action).pluck("facilities.id")
+    Region.facility_regions.where(source_id: facility_ids)
+  end
+
```

Comment:
All the above methods go thru our user_access system, and then find the source objects to determine the matching Region level authorization.